### PR TITLE
Map refresh after layer

### PR DESCRIPTION
**Pull request by index-git**
Part of issue #170

- [x] Tests
- ~~Layman Test Client~~
- [x] Changlelog
- [x] Documentation

Optionally
- Add dependency to doc/dependencies.md

### CODE REVIEW

File: src/layman/layer_map_relation_test.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
+from test import process_client, util as test_util
+from test.data import map as map_data, wfs as data_wfs
+import pytest
+
+from layman import app
+from layman.map.filesystem import thumbnail
+
+
+@pytest.mark.usefixtures('ensure_layman')
+def test_map_refresh_after_layer_change():
```
Comment (jirik) [inline src/layman/layer_map_relation_test.py:10]: I suggest to move it to `layman.publication_relation` package

File: src/layman/util.py [hunk 2]
```diff
@@ -419,3 +421,48 @@ def get_publication_status(workspace, publication_type, publication_name, comple
     else:
         publication_status = 'COMPLETE'
     return publication_status
+
+
+def find_maps_for_layer(layer_workspace, layer_name):
```
[the remark this answers is not in the document]
Comment (jirik) [reply src/layman/util.py:426]: Add unit test

**Review by jirik — CHANGES_REQUESTED** [review]

File: src/layman/map/util.py [hunk 3]
```diff
@@ -308,3 +309,14 @@ def get_map_file_json(username, mapname):
         map_json['user'] = get_map_owner_info(username)
         map_json.pop("groups", None)
     return map_json
+
+
+def find_maps_by_grep(regexp):
+    data_dir = settings.LAYMAN_DATA_DIR
+    grep_cmd = f"grep -r -E '{regexp}' {data_dir}"
```
Comment (jirik) [inline src/layman/map/util.py:316]: Change to `f"grep -r -E '{regexp}' {data_dir}  || [[ $? == 1 ]]"`, see https://unix.stackexchange.com/a/427598 for explanation

This hopefully fixes failed test https://github.com/jirik/layman/runs/2584926010?check_suite_focus=true

File: src/layman/map/filesystem/thumbnail_tasks.py [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+logger = get_task_logger(__name__)
+
+
+@celery_app.task(
+    name='layman.map.filesystem.thumbnail.patch_after_feature_change',
+    bind=True,
+    base=celery_app.AbortableTask
+)
+def patch_after_feature_change(self, workspace, map):
```
[the remark this answers is not in the document]
Comment (index-git) [reply src/layman/map/filesystem/thumbnail_tasks.py:15]: What situations are you missing in commit 6f7ccfd0966731f6682a43b3c5c76c6777faa5bb ?

[the remark this answers is not in the document]
Comment (jirik) [reply src/layman/map/filesystem/thumbnail_tasks.py:15]: Check, that map thumbnail was updated after PATCH Layer

File: src/layman/publication_relation/util.py [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+
+    # verify layer for map
+    gs_ows_url_pattern = fr'^{re.escape(gs_url)}(({layer_workspace}|{layer_wms_workspace})/)?(?:ows|wms|wfs).*$'
+    result_maps = set()
+    for workspace, map in maps:
+        map_json_raw = map_input_file.get_map_json(workspace, map)
+        map_json = map_input_file.unquote_urls(map_json_raw)
+
+        for map_layer in map_json['layers']:
+            layer_url = map_layer.get('url', None)
```
Comment (jirik) [inline src/layman/publication_relation/util.py:36]: Remove redundant None

File: src/layman/publication_relation/util_test.py [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+        yield
+        for map_workspace, map_name in self.maps:
+            process_client.delete_workspace_map(map_workspace, map_name)
+        for workspace, layer in self.layers:
+            process_client.delete_workspace_layer(workspace, layer)
+
+    @staticmethod
+    @pytest.mark.parametrize('layer_workspace, layer_name, expected_maps', [
+        ('testuser1', 'mista', {(workspace, 'map_1')}),
+        ('testuser1', 'faslůghdaůslf', set()),
```
Comment (jirik) [inline src/layman/publication_relation/util_test.py:33]: `faslůghdaůslf` -> `faslughdauslf`

File: test/data/maps/map_1.json [hunk 7]
```diff
@@ -0,0 +1,62 @@
+{
```
Comment (jirik) [inline test/data/maps/map_1.json:1]: Move map json to specific subdir of test/data/maps

File: src/layman/publication_relation/util_test.py [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+from test import process_client
+import pytest
+
+from layman import app
+from . import util
+
+
+class TestFindMapsContainingLayer:
+    workspace = 'test_find_maps_containing_layer_workspace'
+    directory = f'/code/test/data/maps/'
```
Comment (jirik) [inline src/layman/publication_relation/util_test.py:11]: Use more directory specific for this test

File: doc/endpoints.md [hunk 9]
```diff
@@ -15,7 +15,8 @@ The authentication proxy behaves in the same way as in case of [WMS](#web-map-se
...
 - Bounding box and thumbnail of each layer referenced in incoming Insert, Replace, Update, or Delete WFS-T request is updated in [asynchronous chain](async-tasks.md) after the request is finished.
-- Calling WFS-T on a layer when previous asynchronous chain (POST, PATCH or another WFS-T) is still running causes run of WFS-T asynchronous chain after the current one is finished.
+- Calling WFS-T on a layer when previous asynchronous chain of the layer (POST, PATCH or another WFS-T) is still running causes run of WFS-T asynchronous chain after the current one is finished.
+- Calling WFS-T on a layer that is contained by map whose previous asynchronous chain (POST, PATCH or another WFS-T) is still running causes run of map's WFS-T asynchronous chain after the current one is finished.
```
Comment (jirik) [inline doc/endpoints.md:19]: Calling WFS-T on a layer that is contained by map whose previous asynchronous chain (POST, PATCH or another chain caused by WFS-T) is still running causes run of map's asynchronous chain after the current one is finished.

File: doc/rest.md [hunk 10]
```diff
@@ -83,16 +83,18 @@ Publish vector data file as new layer of WMS and WFS.
...
+- save bounding box into PostgreSQL
 - for layers with SLD or none style:
   - publish the table as new layer (feature type) within appropriate WMS workspaces of GeoServer
 - else for layers with QML style:
   - create QGS file on QGIS server filesystem with appropriate style
   - publish the layer on GeoServer through WMS cascade from QGIS server
 - generate thumbnail image
 - publish metadata record to Micka (it's public if and only if read access is set to EVERYONE)
-- save basic information (name, title, access_rights) into PostgreSQL
+- update thumbnail of each [map](models.md#map) that points to this layer.
```
Comment (jirik) [inline doc/rest.md:97]: Remove trailing `.`

**Review by jirik — CHANGES_REQUESTED** [review]

**Review by jirik — APPROVED** [review]

**Pull request merged**